File: be/src/main/java/backend/service/RoomService.java
package backend.service;

import backend.entity.Room;
import java.util.*;

public interface RoomService {

    void delete(long id);
    Room addRoom(Room room);
    Room getById(long id);
    List<Room> getAll();
    List<Room> getFree(Date date1, Date date2);
    int count();
    List<Room> getPage(int page);
}

File: frontend/src/app/room.service.ts
import { Injectable } from '@angular/core';
import {Observable} from 'rxjs';
import {Room} from '../model/Room';
import {HttpClient} from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class RoomService {

  constructor(private http: HttpClient) { }
  public addRoom(room: Room): Observable<any> {
    return this.http.post('api1/room/', room);
  }
  public getAll(): Observable<any> {
    return this.http.get('api1/room/all');
  }
  public delete(id: string): Observable<any> {
    return this.http.delete('api1/room/' + id);
  }
  public getById(id: string): Observable<any> {
    return this.http.get('api1/room/' + id);
  }
  public getFree(date1, date2) {
    const dates = [date1, date2];
    return this.http.post('api1/room/free', dates);
  }
}
